src/test/jtx/mpt.h

```diff
@@ -20,46 +20,177 @@
 #ifndef RIPPLE_TEST_JTX_MPT_H_INCLUDED
 #define RIPPLE_TEST_JTX_MPT_H_INCLUDED
 
-#include <test/jtx/Account.h>
+#include <test/jtx.h>
+#include <test/jtx/ter.h>
+#include <test/jtx/txflags.h>
 
 #include <ripple/protocol/UintTypes.h>
 
 namespace ripple {
 namespace test {
 namespace jtx {
 
-namespace mpt {
-
-/** Issue a MPT with default fields. */
-Json::Value
-create(jtx::Account const& account);
-
-/** Issue a MPT with user-defined fields. */
-Json::Value
-create(
-    jtx::Account const& account,
-    std::uint64_t const maxAmt,
-    std::uint8_t const assetScale,
-    std::uint16_t transferFee,
-    std::string metadata);
-
-/** Destroy a MPT. */
-Json::Value
-destroy(Account const& account, uint192 const& id);
-
-/** Authorize a MPT. */
-Json::Value
-authorize(
-    jtx::Account const& account,
-    uint192 const& issuanceID,
-    std::optional<jtx::Account> const& holder);
-
-/** Set a MPT. */
-Json::Value
-set(jtx::Account const& account,
-    uint192 const& issuanceID,
-    std::optional<jtx::Account> const& holder);
-}  // namespace mpt
+namespace {
+using AccountP = Account const*;
+}
+
+struct MPTConstr
+{
+    std::vector<AccountP> holders = {};
+    PrettyAmount const& xrp = XRP(10'000);
+    PrettyAmount const& xrpHolders = XRP(10'000);
+    bool fund = true;
+    bool close = true;
+};
+
+struct MPTCreate
+{
+    std::optional<std::uint64_t> maxAmt = std::nullopt;
+    std::optional<std::uint8_t> assetScale = std::nullopt;
+    std::optional<std::uint16_t> transferFee = std::nullopt;
+    std::optional<std::string> metadata = std::nullopt;
+    std::optional<std::uint32_t> ownerCount = std::nullopt;
+    bool fund = true;
+    std::uint32_t flags = 0;
+    std::optional<TER> err = std::nullopt;
+};
+
+struct MPTDestroy
+{
+    AccountP issuer = nullptr;
+    std::optional<uint192> id = std::nullopt;
+    std::optional<std::uint32_t> ownerCount = std::nullopt;
+    std::uint32_t flags = 0;
+    std::optional<TER> err = std::nullopt;
+};
+
+struct MPTAuthorize
+{
+    AccountP account = nullptr;
+    AccountP holder = nullptr;
+    std::optional<std::uint32_t> ownerCount = std::nullopt;
+    std::uint32_t flags = 0;
+    std::optional<TER> err = std::nullopt;
+};
+
+struct MPTSet
+{
+    AccountP account = nullptr;
+    AccountP holder = nullptr;
+    std::optional<uint192> id = std::nullopt;
+    std::optional<std::uint32_t> ownerCount = std::nullopt;
+    std::uint32_t flags = 0;
+    std::optional<TER> err = std::nullopt;
+};
+
+class MPTTester
+{
+    Env& env_;
+    Account const& issuer_;
+    std::unordered_map<std::string, AccountP> const holders_;
+    std::optional<std::uint32_t> sequence_;
+    std::optional<uint192> id_;
+    std::optional<uint256> issuanceID_;
```

Comment:
i think this should be called something like `issuanceKey_`

Reply:
And `id_` should be `issuanceID_`, right?

Reply:
i dont think it matters...